### PR TITLE
#207 - Move Qty product import processing to new stock item importer

### PR DESCRIPTION
**Pull request by josh-carter**
### Description
Move Qty Importing to new Stock Item Importer, new extension point to allow for eventual move of product stock import logic to here.

### Fixed Issues (if relevant)
#207 

### Outstanding Tasks
- [x]  Add Extension Point for new Product Import functionality
- [x]  Fix Current Integration Tests
- [x]  Initial Integration Test for new Stock Item Importer

### CODE REVIEW

File: app/code/Magento/CatalogImportExport/Model/Import/Product.php [hunk 1]
```diff
@@ -778,6 +787,7 @@ public function __construct(
         Product\TaxClassProcessor $taxClassProcessor,
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
         \Magento\Catalog\Model\Product\Url $productUrl,
+        StockItemImporterInterface $stockItemImporter,
```
Comment (maghamed) [inline app/code/Magento/CatalogImportExport/Model/Import/Product.php]: currently adding required dependency will introduce Backward Incompatible changes.

It's better to do like we usually do to preserve constructor BC : 
```
 public function __construct(
        \\old arguments 
        StockItemImporterInterface $stockItemImporter = null //add new dependency at the end of the dependencies list
    ) {
        
        $this->stockItemImporter = $stockItemImporter ?: \Magento\Framework\App\ObjectManager::getInstance()->get(
            StockItemImporterInterface::class
        );
```

Comment (josh-carter) [reply app/code/Magento/CatalogImportExport/Model/Import/Product.php]: @maghamed Understood - i've just pushed up a commit to fix this [see here](https://github.com/magento-engcom/msi/pull/211/commits/1af449357e08c0656ce825286d587f99bb17f020)

File: app/code/Magento/CatalogImportExport/Model/Import/Product.php [hunk 2]
```diff
@@ -698,6 +699,13 @@ class Product extends \Magento\ImportExport\Model\Import\Entity\AbstractEntity
      */
     private $catalogConfig;
 
+    /**
+     * Source Item Importer
+     *
+     * @var
```
Comment (maghamed) [inline app/code/Magento/CatalogImportExport/Model/Import/Product.php]: please specify data type for `@var` StockItemImporterInterface

Comment (josh-carter) [reply app/code/Magento/CatalogImportExport/Model/Import/Product.php]: Good spot - [fixed here](https://github.com/magento-engcom/msi/pull/211/commits/290d9193ac1a21423b8a36fad8c3c8115d0ff502)

**Pull request merged**